pkg/ottl/go.mod

```diff
@@ -10,6 +10,7 @@ require (
 	github.com/json-iterator/go v1.1.12
 	github.com/open-telemetry/opentelemetry-collector-contrib/internal/coreinternal v0.105.0
 	github.com/open-telemetry/opentelemetry-collector-contrib/pkg/pdatatest v0.105.0
+	github.com/spaolacci/murmur3 v1.1.0
```

Comment:
[twmb/murmur3](https://github.com/twmb/murmur3) could be an alternative. Initially, I was concerned about the license since it is not listed as BSD-3-Clause. However, after a closer look, it seems acceptable. I did a quick test, and it appears that the hash result from `Sum128()` requires the same byte flip as the current solution. 
I don't see a significant benefit in switching the library, but I am open to change it.

Reply:
One concern I have with `spaolacci/murmur3 - there is an [outstanding issue](https://github.com/spaolacci/murmur3/issues/29), which I believe is the likely cause of some flaky test failures on previous [CI runs](https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/9996255367/job/27667465485?pr=34155) in this PR.

AFAICT, the `twmb/murmur3` is a fork initially created to [specifically fix](https://github.com/spaolacci/murmur3/issues/29#issuecomment-592313738) this issue

Reply:
@robbavey you are right. will change to twmb/murmur3